### PR TITLE
Zub/story/ecom 2233 otto add custom sites model docs

### PR DESCRIPTION
**Pull request by zubair-arbi**
ECOM-2233
@awais786 @aamir-khan @ahsan-ul-haq @tasawernawaz 
- Add basic docs for model `SiteConfiguration` in the app `core`
- Add pre/post populate screen shots of model's django admin

Code changes (PR merged): https://github.com/edx/ecommerce/pull/328
Preview: https://github.com/edx/ecommerce/blob/zub/story/ecom-2233-otto-add-custom-sites-model-docs/docs/partner_config.rst

### CODE REVIEW

**Comment by ahsan-ul-haq** [comment]
:+1:

**Comment by zubair-arbi** [comment]
@clintonb @rlucioni Please review

File: docs/core.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+This is a custom functionality since ``django-oscar`` doesn't offer this kind of configuration.
+We have extended the `Django sites framework <https://docs.djangoproject.com/en/1.8/ref/contrib/sites/>`_ in order to add site-specific configuration.
+The sites framework allows for the mapping of domains to a Site object which, by default, consists an ID and a name.
+
+The multi-tenant implementation has one site per partner.
+
+
+---------------------------------------------
+Site Configuration Model | django Admin
```
Comment (clintonb) [inline docs/core.rst]: Capitalize Django

File: docs/core.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+---------------------------------------------
+
+A new model ``SiteConfiguration`` (shown below in screen shots) will enable the basic theming and payment processor configuration.
+For now you can add/update the custom sites configurations from django admin only.
+
+Sample screen shot of django admin for model ``SiteConfiguration`` **before** populating:
+
+.. image:: images/site_configuration_empty.png
```
Comment (clintonb) [inline docs/core.rst]: The empty state isn't as useful as the completed state you have below.

File: docs/core.rst [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+The sites framework allows for the mapping of domains to a Site object which, by default, consists an ID and a name.
+
+The multi-tenant implementation has one site per partner.
+
+
+---------------------------------------------
+Site Configuration Model | django Admin
+---------------------------------------------
+
+A new model ``SiteConfiguration`` (shown below in screen shots) will enable the basic theming and payment processor configuration.
```
Comment (clintonb) [inline docs/core.rst]: Include an example URL for the page (`http://localhost:8002/admin/partner/partner/`).

**Comment by clintonb** [comment]
Please update the fields in your screenshot:
- Site: ecommerce.stage.edx.org
- LMS URL: https://courses.stage.edx.org
- Theme path: sass/themes/edx.scss

**Comment by clintonb** [comment]
@srpearce do you have some time for a review?

File: docs/index.rst [hunk 4]
```diff
@@ -11,7 +11,8 @@ Contents:
 .. toctree::
    :maxdepth: 2
 
-   fulfillment
+   core.rst
```
Comment (clintonb) [inline docs/index.rst]: You don't need the `.rst` extension.

**Comment by srpearce** [comment]
@clintonb Yep, I can do this today.

**Comment by zubair-arbi** [comment]
@clintonb I have addressed your comments. Please review

File: docs/index.rst [hunk 5]
```diff
@@ -11,6 +11,7 @@ Contents:
 .. toctree::
    :maxdepth: 2
 
+   core
```
Comment (clintonb) [inline docs/index.rst]: partner_config

**Comment by clintonb** [comment]
:+1: (wait for @srpearce to review before merging)

File: docs/index.rst (hunk 5 above)
Comment (zubair-arbi) [inline docs/index.rst]: @clintonb this is part of app `core` so I think it makes sense that it remains `core`. Is that ok?

File: docs/core.rst [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+A new model ``SiteConfiguration`` (shown below in screen shot) will enable the basic theming and payment processor configuration.
+For now you can add/update the custom sites configurations from django admin only (example url: http://localhost:8002/admin/core/siteconfiguration/add/).
+
+Sample screen shot of django admin for model ``SiteConfiguration`` **after** populating:
+
+.. image:: images/site_configuration.png
+    :width: 600px
+    :alt: Populated site configuration model
+
+.. note::  There is **unique together** constraint on the fields **site** and **partner** for model ``SiteConfiguration``.
```
Comment (srpearce) [inline docs/core.rst]: There is a
on the **Site** and **Partner** fields for the `SiteConfiguration` model

File: docs/core.rst [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+The multi-tenant implementation has one site per partner.
+
+
+---------------------------------------------
+Site Configuration Model | Django Admin
+---------------------------------------------
+
+A new model ``SiteConfiguration`` (shown below in screen shot) will enable the basic theming and payment processor configuration.
+For now you can add/update the custom sites configurations from django admin only (example url: http://localhost:8002/admin/core/siteconfiguration/add/).
```
Comment (srpearce) [inline docs/core.rst]: Does this mean that the `SiteConfiguration` model is not currently available, and the screen shot below is just a sneak peek? 

I would rewrite this to say something like this:

Currently, you can only add and update a custom site's configurations by using the Django administration panel. This panel is located at `http://localhost:8002/admin/core/siteconfiguration/add`. _(SP: Be sure to indicate which values in this URL are placeholders.)_

Beginning in September 2015, a new `SiteConfiguration` model will be available. You can use this model to configure the basic theming and payment processor for your site. _(SP: Is this model part of the Django administration panel? If so, we need to be more specific about how the user uses the Django administration panel in the paragraph above - for example, "Currently, you can only add and update a custom's site configurations by using [this tool or method in] the Django administration panel.")_ 

The following image shows the `SiteConfiguration` model in the Django administration panel for a configured site.

File: docs/core.rst [hunk 8]
```diff
@@ -0,0 +1,28 @@
+======================
+Site Configuration
+======================
+
+This is a custom functionality since ``django-oscar`` doesn't offer this kind of configuration.
+We have extended the `Django sites framework <https://docs.djangoproject.com/en/1.8/ref/contrib/sites/>`_ in order to add site-specific configuration.
+The sites framework allows for the mapping of domains to a Site object which, by default, consists an ID and a name.
```
Comment (srpearce) [inline docs/core.rst]: site's framework
**Site** object
consists of

File: docs/index.rst (hunk 5 above)
Comment (clintonb) [inline docs/index.rst]: The documentation layout doesn't need to match the app layout.

Ultimately the documentation will have the following pages:
- Getting started
  - LMS integration
  - OAuth 2.0 setup
- Partner configuration
- Feature gating
- Tests
- Course admin tool

File: docs/core.rst (hunk 7 above)
Comment (zubair-arbi) [inline docs/core.rst]: @srpearce the model `SiteConfiguration` is currently available but we can only update it in django admin.

File: docs/index.rst (hunk 5 above)
Comment (zubair-arbi) [inline docs/index.rst]: @clintonb So do I also need to rename the `rst` file from `core.rst` to `partner_config.rst`?

Comment (clintonb) [inline docs/index.rst]: Correct.

File: docs/core.rst (hunk 7 above)
Comment (srpearce) [inline docs/core.rst]: OK, thanks. The "will" in the first sentence and the "for now" in the second were confusing. In that case, I'd suggest the following.

To add and update a custom site's configurations, including the basic theming and payment processor, use the `SiteConfiguration` model in the Django administration panel. This panel is located at http://localhost:8002/admin/core/siteconfiguration/add. _(SP: Be sure to indicate which values in this URL are placeholders.)_

The following image shows...

**Comment by zubair-arbi** [comment]
@srpearce @clintonb I have addressed you comments and its ready for final review.

**Comment by srpearce** [comment]
@zubair-arbi Aargh - GitHub seems to have eaten the comment I made a few minutes ago (as well as your reply to an earlier comment). I had suggested a new revision based on your feedback that `SiteConfiguration` is already available. I'll re-add that comment in just a minute...

**Comment by zubair-arbi** [comment]
@srpearce No need to re-add comment. I got that and now updating the PR.

**Comment by srpearce** [comment]
Oh, great. Let me know when you're done and I'll give it a final check.

**Comment by zubair-arbi** [comment]
@srpearce its ready!

File: docs/partner_config.rst [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+To add and update a custom site's configurations, including the basic theming and payment processor, use the ``SiteConfiguration`` model in the `Django administration panel <http://localhost:8002/admin/core/siteconfiguration/add>`_.
+This panel is located at http://localhost:8002/admin/core/siteconfiguration/.
+
+The following image shows the ``SiteConfiguration`` model in the Django administration panel for a configured site.
+
+.. image:: images/site_configuration.png
+    :width: 600px
+    :alt: Populated site configuration model
+
+.. note::  There is a **unique together** constraint on the **site** and **partner** fields for ``SiteConfiguration`` model.
```
Comment (srpearce) [inline docs/partner_config.rst]: for the `SiteConfiguration`

**Comment by srpearce** [comment]
@zubair-arbi Great! Just one more thing (I think I missed this before). Otherwise, :+1:

**Pull request merged**